optaplanner-core/src/test/java/org/optaplanner/core/impl/heuristic/selector/common/nearby/BetaDistributionNearbyRandomTest.java

```diff
@@ -40,14 +40,12 @@ public void nextIntUniform() {
         Random random = mock(Random.class);
         NearbyRandom nearbyRandom = new BetaDistributionNearbyRandom(1.0, 1.0);
 
-        when(random.nextDouble()).thenReturn(0.0);
-        assertEquals(0, nearbyRandom.nextInt(random, 500));
-        when(random.nextDouble()).thenReturn(1.0 / 500.0);
-        assertEquals(1, nearbyRandom.nextInt(random, 500));
-        when(random.nextDouble()).thenReturn(2.0 / 500.0);
-        assertEquals(2, nearbyRandom.nextInt(random, 500));
-        when(random.nextDouble()).thenReturn(3.0 / 500.0);
-        assertEquals(3, nearbyRandom.nextInt(random, 500));
+        double roundingConstant = 0.000001; // normalizes calculation against over-rounding error
+        for (int i = 0; i < 500; i++) {
```

Comment:
I prefer hard coded numbers in unit tests, instead of dynamic code to validate all numbers. Because there's a high risk that that dynamic test code is written under the same logic as the production code, victim to the same bugs. Hard coded numbers on the other hand, predicted in advance on paper (so written without using the production code or even a bit of code to calculate the result), are a better approach in my opinion.

So I am going to revert this class to the original code.